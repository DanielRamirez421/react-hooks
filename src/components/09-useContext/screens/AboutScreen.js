import React, { Fragment, useContext } from 'react';
import { UserContext } from '../context/UserContext';

export const AboutScreen = () => {

  const { user, setUser } = useContext( UserContext ); 

  const handleClick = () => setUser({});

  return (
    <Fragment>
      <h1>About Screen</h1>
      <hr />

      <pre>
        { JSON.stringify( user, null, 3 ) }
      </pre>

      <button
        className='btn alert-warning'
        onClick={ handleClick }>
        LogOut
      </button>

    </Fragment>
    );
};
