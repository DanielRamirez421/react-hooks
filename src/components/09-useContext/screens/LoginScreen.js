import React, { useContext } from 'react';
import { UserContext } from '../context/UserContext';

export const LoginScreen = () => {

  const { setUser } = useContext(UserContext);
  const userExample = { id: 123, name: 'Daniel', username: 'damtrax' }

  return (
    <div>
      <h1>Login Screen</h1>
      <hr />

      <button
        className='btn btn-outline-primary'
        onClick={ () => setUser(userExample) }>
        SetUser
      </button>
    </div>
    );
};
