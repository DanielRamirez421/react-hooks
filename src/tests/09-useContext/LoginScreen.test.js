import { mount } from "enzyme";
import { UserContext } from "../../components/09-useContext/context/UserContext";
import { LoginScreen } from "../../components/09-useContext/screens/LoginScreen";

describe('LoginScreen component tests', () => {

  const setUser = jest.fn();
  const wrapper = mount( 
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen /> 
    </UserContext.Provider>
  );

  test('should show component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call setUser fnc', () => {
    // wrapper.find( 'button' ).simulate('click'); // Another way to simulate click
    wrapper.find('button').prop('onClick')();
    expect(setUser).toHaveBeenCalledTimes(1);
    expect(setUser).toHaveBeenCalledWith({ id: 123, name: 'Daniel', username: 'damtrax' });
  });

});