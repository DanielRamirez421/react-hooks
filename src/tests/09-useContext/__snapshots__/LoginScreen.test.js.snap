// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen component tests should show component 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    SetUser
  </button>
</div>
`;
